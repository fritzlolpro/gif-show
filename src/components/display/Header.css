.header-menu {
  width: 100%;
  background-color: #fff;
  height: 40px;
  position: fixed;
  opacity: .8;
  z-index: 1;
}
.header-menu:hover a,
.header-menu:hover button,
.header-menu:hover label {
  color: #1d77dd;
}
.header-menu:hover a i,
.header-menu:hover button i,
.header-menu:hover label i {
  color: #1d77dd;
}
.header-menu a {
  text-decoration: none;
  float: left;
  position: relative;
  left: 5%;
  margin-top: 7px;
}
.header-menu i {
  color: #4a90e2;
  vertical-align: middle;
  margin-right: 8px;
}
.header-menu label {
  font-size: 24px;
  color: #4a90e2;
  display: inline-block;
  text-transform: uppercase;
  font-weight: 600;
  vertical-align: middle;
}
.header-menu button {
  float: right;
  right: 5%;
  padding: 0;
  border: 0;
  position: relative;
  margin-top: 7px;
  background: none;
  cursor: pointer;
  outline: none;
  -webkit-transform: rotate(90deg);
          transform: rotate(90deg);
}
.header-menu button i {
  font-weight: 600;
  width: 100%;
}
.header-menu button:active i {
  -webkit-transform: rotate(-45deg);
          transform: rotate(-45deg);
}
