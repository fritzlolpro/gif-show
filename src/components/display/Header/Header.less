.header-menu {
    width: 100%;
    background-color: #fff;
    height: 40px;
    position: fixed;
    opacity: .8;
    z-index: 1;
    &:hover {
        a,
        button,
        label {
            color: #1d77dd;
            i {
                color: #1d77dd;
            }
        }
    }
    a {
        text-decoration: none;
        float: left;
        position: relative;
        left: 5%;
        margin-top: 7px;
    }
    i {
        color: #4a90e2;
        vertical-align: middle;
        margin-right: 8px;
    }
    label {
        font-size: 24px;
        color: #4a90e2;
        display: inline-block;
        text-transform: uppercase;
        font-weight: 600;
        vertical-align: middle;
    }
    button {
        float: right;
        right: 5%;
        padding: 0;
        border: 0;
        position: relative;
        margin-top: 7px;
        background: none;
        cursor: pointer;
        outline: none;
        transform: rotate(90deg);
        i {
            font-weight: 600;
            width: 100%
        }
        &:active {
            i {
                transform: rotate(-45deg);
            }
        }
    }
}