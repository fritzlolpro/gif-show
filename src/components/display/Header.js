import React from 'react';
import Refresher from '../control/Refresher'
import {Route, Link} from "react-router-dom";
import './Header.css';

const Header = ({handleRefresh}) => {
    return (
        <div className='header-menu'>
            <Link to='/'>
               <i className="material-icons">public</i>
               <label>giphlar</label>
            </Link>


               <Route exact path='/' render={() => (
                   <Refresher onRefresh={handleRefresh}/>
               )}/>

        </div>
    )
}




export default Header