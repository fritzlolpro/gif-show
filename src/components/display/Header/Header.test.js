import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({adapter: new Adapter()})

import Header from './Header'

describe('Header', () => {

    it('Defined', () => {
        expect(Header).toBeDefined();
    });

    it('Renders', () => {
        const tree = shallow(<Header/>);
        expect(tree).toMatchSnapshot();
    });


})