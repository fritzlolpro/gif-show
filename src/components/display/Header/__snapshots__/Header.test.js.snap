// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
>
          <i
                    className="material-icons"
          >
                    public
          </i>
          <label>
                    giphlar
          </label>
</Link>,
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
      ],
      "className": "header-menu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <i
              className="material-icons"
>
              public
</i>,
            <label>
              giphlar
</label>,
          ],
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "public",
              "className": "material-icons",
            },
            "ref": null,
            "rendered": "public",
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "giphlar",
            },
            "ref": null,
            "rendered": "giphlar",
            "type": "label",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
>
            <i
                        className="material-icons"
            >
                        public
            </i>
            <label>
                        giphlar
            </label>
</Link>,
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
        ],
        "className": "header-menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="material-icons"
>
                public
</i>,
              <label>
                giphlar
</label>,
            ],
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "public",
                "className": "material-icons",
              },
              "ref": null,
              "rendered": "public",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "giphlar",
              },
              "ref": null,
              "rendered": "giphlar",
              "type": "label",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
